src/main/java/bank/PaymentService.java

```diff
@@ -5,11 +5,23 @@
  */
 public class PaymentService {
 
-    public void transferMoney(Account accountOne, Account accountTwo, int moneyToTransfer) {
+    public void transferMoney(Account accountOne, Account accountTwo, Instrument moneyToTransfer) {
+
+//        accountOne.setBalance(accountOne.getBalance()-moneyToTransfer);
+//        if(accountOne.getBalance() < -500){
+//            throw new IllegalArgumentException("I'm very sorry, but you don't have enough money...");
+//        }
+//        accountTwo.setBalance(accountTwo.getBalance()+moneyToTransfer);
 
-        accountOne.setBalance(accountOne.getBalance()-moneyToTransfer);
-        accountTwo.setBalance(accountTwo.getBalance()+moneyToTransfer);
-    }
 
 
+
+        accountOne.getBalance().setAmount(accountOne.getBalance().getAmount()-moneyToTransfer.getAmount());
+        if(accountOne.getBalance().getAmount() < -500){
```

Comment:
Look, here we will transfer money even if there will be amount less than 500. 
It would be nice to have test scenario which will check if amount was changed, even when such exception was thrown. :)